cmd/kubeadm/app/apis/kubeadm/types.go

```diff
@@ -43,9 +43,9 @@ type MasterConfiguration struct {
 }
 
 type API struct {
-	AdvertiseAddresses []string
-	ExternalDNSNames   []string
-	Port               int32
+	BindAddress      string
```

Comment:
That's tested and it's working.

> 0.0.0.0 says "listen on all addresses."

That's not true -- 0.0.0.0 stands for the IP address of the default network interface